### DISCUSSION TITLE
When/how should we publish Material-UI 5.

### DISCUSSION THREAD

**Discussion by bertearazvan** [DHI/react-components#280] [Show and tell]
This is the place where I would like to discuss how to best make the change of migrating the "react-components-lab" to material-ui 5.

Probably the theme.5 cannot be used in material-ui 4 projects and migrating a whole app to Material-UI 5 is not really desirable, therefore, I think this is best to be used when creating new projects. By it being used, we can spot different issues, like the theme or specific components. _So please, use it when feasible._

A way to go around this is to publish a major version `2` of the "react-components-lab" library, while if needed we can improve the Material-UI.4 through the version `1.x.y`. Here is something I am not sure of: 
- If we have version `1.12.6` and we publish `2.0.0`, will we be able to publish `1.13.0` afterward?

This way we make sure that people can still use Material-UI.4 and we can still push fixes here and there when needed.

**For a more technical overview and changes**, you can take a look at https://github.com/DHI/react-components/pull/270

I would like to hold a knowledge-sharing session on how to migrate to material-UI 5, how does the new structure look like, and how to use Styled components further on.

I would like to hear your considerations if any @FranzThomsen1089 @nfour @goneale @rihr @rosudhi @elbys @j08lue

**Comment by rihr** [comment]
Using npm/registry tags could worth considering. If I understand correctly, that would allow to publish and maintain both v4 and v5 packages.
Would also need to decide how code would be maintained for a v4 package, if we were to do that

**Comment by bertearazvan** [comment]
So it is confirmed, the Mui.v5 theme is not working when applied on Material-UI.v4 components.
